flaml/data.py

```diff
@@ -197,8 +197,21 @@ def fit_transform(self, X, y, task):
             for column in X.columns:
                 # sklearn\utils\validation.py needs int/float values
                 if X[column].dtype.name == 'datetime64[ns]':
+                    X[f'year_{column}'] = X[column].dt.year
```

Comment:
we need to make sure these newly added columns do not exist in the original dataframe. If they exist, we shouldn't overwrite them.

Reply:
What about using a random prefix, for each column name ?